kubectl-volsync/cmd/migration_create_test.go

```diff
@@ -0,0 +1,281 @@
+		err = mc.parseCLI(cmd)
+		Expect(err).To(HaveOccurred())
+	})
+
+	It("Verify namespace creation with no pre-existing namespace", func() {
+		mrd.Namespace = mc.Namespace
+		ns, err := mc.ensureNamespace(cmd.Context())
+		Expect(err).ToNot(HaveOccurred())
+		Expect(ns).NotTo(BeNil())
+
+	})
+
+	It("Verify PVC creation with/without pre-existing PVC", func() {
+		PVC, err := mc.getDestinationPVC(cmd.Context())
+		Expect(err).NotTo(HaveOccurred())
+		mc.PVC = PVC
+
+		dest, err := mc.newMigrationRelationshipDestination()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(dest).ToNot(BeNil())
+		mc.mr.data.Destination = dest
+
+		ns, err = mc.ensureNamespace(cmd.Context())
+		Expect(err).ToNot(HaveOccurred())
+
+		destPVC, err := mc.getDestinationPVC(cmd.Context())
+		Expect(err).NotTo(HaveOccurred())
+		if destPVC != nil {
+			dest := mc.getDestination(cmd.Context())
+			if dest != nil {
+				err = mc.deleteDestination(cmd.Context())
+				Expect(err).NotTo(HaveOccurred())
+			}
+			err = mc.deleteDestinationPVC(cmd.Context())
+			Expect(err).NotTo(HaveOccurred())
+		}
+
+		mc.PVC, err = mc.ensureDestPVC(cmd.Context())
+		Expect(err).ToNot(HaveOccurred())
+		Expect(mc.PVC).ToNot(BeNil())
+
+		// PVC creation should fail if PVC already exists
+		mc.PVC, err = mc.ensureDestPVC(cmd.Context())
+		Expect(err).NotTo(HaveOccurred())
+		Expect(mc.PVC).NotTo(BeNil())
+
+	})
+
+	It("Verify migration destination creation with/without pre-existing migration destination", func() {
+		PVC, err := mc.getDestinationPVC(cmd.Context())
+		Expect(err).NotTo(HaveOccurred())
+		mc.PVC = PVC
+
+		mrd, err := mc.newMigrationRelationshipDestination()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(mrd).ToNot(BeNil())
+		mc.mr.data.Destination = mrd
+
+		ns, err = mc.ensureNamespace(cmd.Context())
+		Expect(err).ToNot(HaveOccurred())
+
+		mc.PVC, err = mc.getDestinationPVC(cmd.Context())
+		Expect(err).NotTo(HaveOccurred())
+		if mc.PVC == nil {
+			mc.PVC, _ = mc.ensureDestPVC(cmd.Context())
+		}
+		Expect(mc.PVC).ToNot(BeNil())
+
+		dest := mc.getDestination(cmd.Context())
+		if dest != nil {
+			err = mc.deleteDestination(cmd.Context())
+			Expect(err).NotTo(HaveOccurred())
+		}
+		rd, err := mc.ensureReplicationDestination(cmd.Context())
+
+		Expect(err).ToNot(HaveOccurred())
+		Expect(rd).ToNot(BeNil())
+
+		// Create destination should fail if destination exists
+		rd, err = mc.ensureReplicationDestination(cmd.Context())
+		Expect(err).To(HaveOccurred())
+		Expect(rd).To(BeNil())
+	})
+})
+
+func migrationCmdArgsSet(cmd *cobra.Command, migrationCmdArgs map[string]string) error {
+	for i, v := range migrationCmdArgs {
+		err := cmd.Flags().Set(i, v)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (mc *migrationCreate) deleteNamespace(ctx context.Context, ns *corev1.Namespace) error {
+	if err := mc.clientObject.Delete(ctx, ns); err != nil {
+		return err
+	}
+	err := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
```

Comment:
You don't need to wait for completion of `Delete()`, and it may never complete due to the test API server.

Deletions can just be written as:
```go
Expect(k8sClient.Delete(ctx, obj)).To(Succeed())
```

Reply:
But the subsequent tests will fail if the deletion status is still "Terminating" when a new test starts.

For ex:
```

I0123 15:49:21.674848  131341 migration_create.go:263] Namespace: "dest" is found, proceeding with the same
• Failure [0.228 seconds]
migration
/home/vharihar/project/volsync/kubectl-volsync/cmd/migration_create_test.go:20
  Ensure replicationdestination creation [It]
  /home/vharihar/project/volsync/kubectl-volsync/cmd/migration_create_test.go:138

  Unexpected error:
      <*errors.StatusError | 0xc0002e0140>: {
          ErrStatus: {
              TypeMeta: {Kind: "", APIVersion: ""},
              ListMeta: {
                  SelfLink: "",
                  ResourceVersion: "",
                  Continue: "",
                  RemainingItemCount: nil,
              },
              Status: "Failure",
              Message: "persistentvolumeclaims \"volsync\" is forbidden: unable to create new content in namespace dest because it is being terminated",
              Reason: "Forbidden",
              Details: {
                  Name: "volsync",
                  Group: "",
                  Kind: "persistentvolumeclaims",
                  UID: "",
                  Causes: [
                      {
                          Type: "NamespaceTerminating",
                          Message: "namespace dest is being terminated",
                          Field: "metadata.namespace",
                      },
                  ],
                  RetryAfterSeconds: 0,
              },
              Code: 403,
          },
      }
      persistentvolumeclaims "volsync" is forbidden: unable to create new content in namespace dest because it is being terminated
  occurred
```
 Is it not a good idea to wait till completion and start a fresh test?